website_membership_random_order/models/membership_line.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2021 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import random
+
+from odoo import models
+
+
+class MembershipLine(models.Model):
+    _inherit = "membership.membership_line"
+
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        """If we detect the injected context, and only when the search is done with
+        limit (as there are 2 calls to search in the controller), we search for all
+        the results, randomize them, and return them. We can't do slices, as if not, we
+        won't be able to show all the records due to the pager and each time returning
+        random results.
+        """
+        if self.env.context.get("random_membership_line_order") and limit:
+            record_ids = super().search(args, limit=None).ids
```

Comment:
remove limit argument

Reply:
Well, it's on purpose to highlight that no limit is applied. Please read docstring.